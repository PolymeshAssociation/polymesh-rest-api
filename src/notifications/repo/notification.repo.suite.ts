/* istanbul ignore file */

import { NotificationRepo } from '~/notifications/repo/notifications.repo';
import { NotificationStatus } from '~/notifications/types';

export const testNotificationRepo = async (notificationRepo: NotificationRepo): Promise<void> => {
  let id: number;

  describe('method: create', () => {
    it('should create a notification', async () => {
      const notification = await notificationRepo.create({
        status: NotificationStatus.Active,
        eventId: 0,
        subscriptionId: 1,
        nonce: 1,
        triesLeft: 3,
      });
      id = notification.id;

      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { createdAt: _, ...params } = notification;
      expect(params).toMatchSnapshot();
    });
  });

  describe('method: findById', () => {
    it('should find the created notification', async () => {
      const notification = await notificationRepo.findById(id);

      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { createdAt: _, ...params } = notification!;

      expect(params).toMatchSnapshot();
    });
  });

  describe('method: update', () => {
    it('should update the notification', async () => {
      await notificationRepo.update(id, { status: NotificationStatus.Acknowledged });

      const notification = await notificationRepo.findById(id);

      expect(notification?.status).toEqual(NotificationStatus.Acknowledged);
    });
  });
};
