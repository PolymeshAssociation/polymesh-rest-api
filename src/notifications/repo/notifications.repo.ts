import { NotificationModel } from '~/notifications/model/notification.model';
import { testNotificationRepo } from '~/notifications/repo/notification.repo.suite';
import { NotificationParams } from '~/notifications/types';

export abstract class NotificationRepo {
  public static readonly type = 'Notification';

  public abstract create(params: NotificationParams): Promise<NotificationModel>;
  public abstract findById(id: number): Promise<NotificationModel | undefined>;
  public abstract update(
    id: number,
    params: Partial<NotificationParams>
  ): Promise<NotificationModel>;

  /**
   * a set of tests implementers should pass
   */
  public static async test(repo: NotificationRepo): Promise<void> {
    return testNotificationRepo(repo);
  }
}
