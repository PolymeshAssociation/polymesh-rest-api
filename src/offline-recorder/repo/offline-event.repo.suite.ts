import { TopicName } from '~/common/utils/amqp';
import { OfflineEventModel } from '~/offline-recorder/model/offline-event.model';
import { OfflineEventRepo } from '~/offline-recorder/repo/offline-event.repo';

const name = TopicName.Signatures;
const body = { id: 'abc', memo: 'offline suite test' };

export const testOfflineEventRepo = async (offlineRepo: OfflineEventRepo): Promise<void> => {
  let event: OfflineEventModel;

  describe('method: recordEvent', () => {
    it('should record an event', async () => {
      event = await offlineRepo.recordEvent(name, body);
      expect(event).toMatchSnapshot();
    });
  });
};
