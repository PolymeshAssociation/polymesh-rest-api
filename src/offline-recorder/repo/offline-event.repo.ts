import { AnyModel } from '~/offline-recorder/model/any.model';
import { OfflineEventModel } from '~/offline-recorder/model/offline-event.model';
import { testOfflineEventRepo } from '~/offline-recorder/repo/offline-event.repo.suite';

export abstract class OfflineEventRepo {
  public static type = 'OfflineEvent';

  public abstract recordEvent(name: string, body: AnyModel): Promise<OfflineEventModel>;

  /**
   * a set of tests implementers should pass
   */
  public static async test(repo: OfflineEventRepo): Promise<void> {
    return testOfflineEventRepo(repo);
  }
}
