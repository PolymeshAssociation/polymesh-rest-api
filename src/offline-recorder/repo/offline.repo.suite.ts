import { TopicName } from '~/common/utils/amqp';
import { OfflineEventModel } from '~/offline-recorder/model/event.model';
import { OfflineRepo } from '~/offline-recorder/repo/offline.repo';

const name = TopicName.Submissions;
const body = { id: 'abc' };

export const testOfflineRepo = async (offlineRepo: OfflineRepo): Promise<void> => {
  let event: OfflineEventModel;

  describe('method: recordEvent', () => {
    it('should create a user', async () => {
      event = await offlineRepo.recordEvent(name, body);
      expect(event).toMatchSnapshot();
    });
  });
};
