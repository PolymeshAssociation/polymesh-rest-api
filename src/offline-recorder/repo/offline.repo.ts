import { OfflineEventModel } from '~/offline-recorder/model/event.model';
import { testOfflineRepo } from '~/offline-recorder/repo/offline.repo.suite';

export abstract class OfflineRepo {
  public static type = 'Offline';

  public abstract recordEvent(
    name: string,
    body: Record<string, unknown>
  ): Promise<OfflineEventModel>;

  /**
   * a set of tests implementers should pass
   */
  public static async test(repo: OfflineRepo): Promise<void> {
    return testOfflineRepo(repo);
  }
}
