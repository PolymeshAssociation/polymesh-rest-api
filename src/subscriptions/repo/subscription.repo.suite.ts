/* istanbul ignore file */

import { AppNotFoundError } from '~/common/errors';
import { EventType } from '~/events/types';
import { SubscriptionRepo } from '~/subscriptions/repo/subscription.repo';
import { SubscriptionStatus } from '~/subscriptions/types';

export const testSubscriptionRepo = async (subscriptionRepo: SubscriptionRepo): Promise<void> => {
  let id: number;

  describe('method: create', () => {
    it('should create a subscription', async () => {
      const subscription = await subscriptionRepo.create({
        status: SubscriptionStatus.Inactive,
        eventType: EventType.TransactionUpdate,
        eventScope: '',
        webhookUrl: 'http://example.com',
        legitimacySecret: 'someSecret',
        ttl: 10,
        triesLeft: 10,
        nextNonce: 3,
        createdAt: new Date('1987-10-14'),
      });

      id = subscription.id;
      expect(subscription).toMatchSnapshot();
    });
  });

  describe('method: findById', () => {
    it('should find the created notification', async () => {
      const subscription = await subscriptionRepo.findById(id);

      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { createdAt: _, ...params } = subscription!;

      expect(params).toMatchSnapshot();
    });

    it('should throw an error for an ID that does not exist', async () => {
      return expect(subscriptionRepo.findById(-1)).rejects.toThrow(AppNotFoundError);
    });
  });

  describe('method: update', () => {
    it('should update the notification', async () => {
      await subscriptionRepo.update(id, { status: SubscriptionStatus.Active });

      const subscription = await subscriptionRepo.findById(id);

      expect(subscription?.status).toEqual(SubscriptionStatus.Active);
    });
  });

  describe('method: findAll', () => {
    it('should return all of the subscriptions', async () => {
      const subscriptions = await subscriptionRepo.findAll();

      expect(subscriptions.length).toEqual(1);
      expect(subscriptions[0].id).toEqual(id);
    });
  });

  describe('method: incrementNonces', () => {
    it('should increment nonces', async () => {
      await subscriptionRepo.incrementNonces([id]);

      const subscription = await subscriptionRepo.findById(id);

      expect(subscription?.nextNonce).toEqual(4);
    });
  });
};
