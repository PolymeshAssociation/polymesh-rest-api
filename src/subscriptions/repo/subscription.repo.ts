import { SubscriptionModel } from '~/subscriptions/models/subscription.model';
import { testSubscriptionRepo } from '~/subscriptions/repo/subscription.repo.suite';
import { SubscriptionParams } from '~/subscriptions/types';

export abstract class SubscriptionRepo {
  public static readonly type = 'Subscription';

  public abstract create(
    params: Omit<SubscriptionModel, 'id' | 'isExpired'>
  ): Promise<SubscriptionModel>;

  public abstract findAll(): Promise<SubscriptionModel[]>;

  public abstract findById(id: number): Promise<SubscriptionModel | undefined>;

  public abstract update(
    id: number,
    params: Partial<SubscriptionParams>
  ): Promise<SubscriptionModel>;

  public abstract incrementNonces(id: number[]): Promise<void>;

  /**
   * a set of tests implementers should pass
   */
  public static async test(repo: SubscriptionRepo): Promise<void> {
    return testSubscriptionRepo(repo);
  }
}
