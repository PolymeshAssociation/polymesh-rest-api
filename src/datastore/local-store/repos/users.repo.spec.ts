import { LocalUserRepo } from '~/datastore/local-store/repos/users.repo';
import { UsersRepo } from '~/users/repo/user.repo';

describe(`LocalUserRepo does not meet ${UsersRepo.type} test suite requirements`, () => {
  const repo = new LocalUserRepo();

  UsersRepo.test(repo);
});
