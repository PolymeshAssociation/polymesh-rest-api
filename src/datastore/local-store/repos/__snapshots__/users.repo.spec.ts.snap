// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalUserRepo does not meet User test suite requirements method: createUser should create a user 1`] = `
{
  "id": "1",
  "name": "Alice",
}
`;

exports[`LocalUserRepo does not meet User test suite requirements method: findByName should find the created user 1`] = `
{
  "id": "1",
  "name": "Alice",
}
`;
